packages/shared/test-utils/src/mocks.ts

```
@@ -0,0 +1,100 @@
+      events.get(event)!.push(handler)
+    }),
+    rooms: new Set<string>(),
+  }
+}
+
+// Improved type safety for job operations
+export function createTypedMockQueue<T = any>(): IQueueService {
+  const jobs: Array<Job<T>> = []
+
+  return {
+    addJob: vi.fn(async <TData>(queue: string, name: string, data: TData, options?: unknown) => {
```

ESLint

[error] 'options' is defined but never used. Allowed unused args must match /^_/u.

(@typescript-eslint/no-unused-vars)